apps/ledger-live-mobile/src/screens/FirmwareUpdate/useUpdateFirmwareAndRestoreSettings.ts

```diff
@@ -193,14 +308,90 @@ export const useUpdateFirmwareAndRestoreSettings = ({
     triggerUpdate,
     updateActionState.step,
     updateStep,
+    restoreAppsState.error,
+    restoreAppsState.opened,
   ]);
 
+  const hasReconnectErrors = useMemo(
+    () =>
+      reconnectDeviceErrors.some(
+        err => connectManagerState.error instanceof err,
+      ) ||
+      reconnectDeviceErrors.some(
+        err => staxFetchImageState.error instanceof err,
+      ) ||
+      reconnectDeviceErrors.some(
+        err => staxLoadImageState.error instanceof err,
+      ) ||
+      reconnectDeviceErrors.some(
+        err => restoreAppsState.error instanceof err,
+      ) ||
+      reconnectDeviceErrors.some(
+        err => installLanguageState.error instanceof err,
```

Comment:
There is no way to do all those checks inside only 1 `reconnectDeviceErrors.some(...)` ?